src/kit.ts

```diff
@@ -580,6 +595,70 @@ async function collectDevBatVars(devbat: string, args: string[], major_version:
   return vars;
 }
 
+/**
+ * Gets the environment variables set by a shell script.
+ * @param kit The kit to get the environment variables for
+ */
+export async function getShellScriptEnvironment(kit: Kit): Promise<Map<string, string>|undefined> {
+  console.assert(kit.environmentVariablesShellScript);
+  const fname = Math.random().toString() + (process.platform == 'win32' ? '.bat' : '.sh');
+  const script_fname = `vs-cmt-${fname}`;
+  const envfname = script_fname + '.env';
+  const script_path = path.join(paths.tmpDir, script_fname);
+  const env_path = path.join(paths.tmpDir, envfname); // path of temp file in which the script writes the env vars to
+  const args = kit.environmentVariablesShellScriptArgs ? kit.environmentVariablesShellScriptArgs.map(arg => `"${arg}"`).join(" ") : "";
+
+  const script = [];
+  if (process.platform == 'win32') { // windows
+    if (kit.environmentVariablesShellScriptDirectory) {
+      script.push(`cd /d "${kit.environmentVariablesShellScriptDirectory}"`); // switch to directory if set
+    }
+    script.push(`call "${kit.environmentVariablesShellScript}" ${args}`); // call the user batch script
+    script.push(`set >> ${env_path}`); // write env vars to temp file
+    }
+  else { // linux based
```

Comment:
`else` goes on the previous line after the `}`

Reply:
Thats true